qualtran/surface_code/t_factory.py

```diff
@@ -0,0 +1,52 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import math
+
+from attrs import field, frozen
+
+from qualtran.surface_code.algorithm_summary import AlgorithmSummary
+from qualtran.surface_code.magic_state_factory import MagicStateFactory
+
+
+@frozen
+class TFactory(MagicStateFactory):
+    """A magic state factory for T states.
+
+    Attributes:
+        num_qubits: Number of qubits used by the factory.
+        duration: Time taken by the factory to produce T states.
+        t_states_rate: Number of T states per production cycle.
+        reference: Source of these estimates.
+    """
+
+    num_qubits: int
+    generation_cycle_duration_ns: float = field(repr=lambda x: f'{x:g}')
```

Comment:
I would strongly advocate for not mixing physical units into this definition. Isn't that the point of your other class to contain the physical attributes of a device/scheme?

Reply:
ptal. I renamed the parameter to better reflect what they are. none of them are physical